File: RecyclerSIngleSelectionPractice/app/src/main/java/com/gandan/android/recyclersingleselectionpractice/RecyclerAdapter.java
package com.gandan.android.recyclersingleselectionpractice;

import android.content.Context;
import android.graphics.Color;
import android.support.annotation.NonNull;
import android.support.constraint.ConstraintLayout;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;
import java.util.List;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.Holder> {

    Context context;
    List<Item> contact;
    boolean single;

    public RecyclerAdapter(Context context, List<Item> contact, boolean single){
        this.context = context;
        this.contact = contact;
        this.single = single;
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.reycler_item, parent, false);
        return new Holder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final Holder holder, final int position) {
        final Item item = contact.get(position);

        holder.textName.setText(item.getName());
        if(item.isSelect()){
            holder.textName.setTextColor(Color.parseColor("#01cda6"));
            holder.textName.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                   Toast.makeText(context, "이미 선택됨", Toast.LENGTH_SHORT).show();
                }
            });
        } else {
            holder.textName.setTextColor(Color.parseColor("#1e1e1e"));
            holder.textName.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    RxEventBus.getInstance().sendEvent(item);
                }
            });
        }
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

    @Override
    public int getItemCount() {
        return contact.size();
    }

    class Holder extends RecyclerView.ViewHolder{

        TextView textName;
        ConstraintLayout layoutItem;
        boolean select;

        public Holder(View itemView) {
            super(itemView);
            textName = itemView.findViewById(R.id.txt_item_name);
            layoutItem = itemView.findViewById(R.id.layout_item);
        }
    }
}

File: DiffUtil/app/src/main/java/com/gandan/android/diffutil/RecyclerAdapter.kt
package com.gandan.android.diffutil

import android.content.Context
import android.graphics.Color
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.selection.ItemDetailsLookup
import androidx.recyclerview.selection.SelectionTracker
import androidx.recyclerview.widget.RecyclerView

class RecyclerAdapter(var dummyList: ArrayList<DataModels.DummyModel>) : RecyclerView.Adapter<RecyclerAdapter.RecyclerHolder>() {

    var tracker : SelectionTracker<Long>? = null

    init {
        setHasStableIds(true)
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerHolder {
        /*TODO("not implemented") //To change body of created functions use File | Settings | File Templates.*/
        var view  = LayoutInflater.from(parent.context).inflate(R.layout.recyclerview_item, parent, false)
        return RecyclerHolder(view)
    }

    override fun getItemCount(): Int {
        /*TODO("not implemented") //To change body of created functions use File | Settings | File Templates.*/
        return dummyList.size
    }

    override fun onBindViewHolder(holder: RecyclerHolder, position: Int) {
        //TODO("not implemented") //To change body of created functions use File | Settings | File Templates.)
        holder.nameTxtView.text = dummyList[position].name
        tracker!!.let {
            holder.bind(dummyList[position].name, it.isSelected(position.toLong()))
        }

    }



    inner class RecyclerHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        var nameTxtView = itemView.findViewById<TextView>(R.id.itemTextView)!!
        var backGroundLinear = itemView.findViewById<LinearLayout>(R.id.itemLinear)!!

        fun bind(name : String?, isActivated: Boolean = false) {
            nameTxtView.text = name
            itemView.isActivated = isActivated
            if (isActivated) backGroundLinear.setBackgroundColor(Color.BLUE) else backGroundLinear.setBackgroundColor(Color.WHITE)

        }

        fun getItemDetails() : ItemDetailsLookup.ItemDetails<Long> =
                object : ItemDetailsLookup.ItemDetails<Long>() {
                    override fun getPosition(): Int = adapterPosition
                    override fun getSelectionKey(): Long? = itemId
                }
    }
}